packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -667,64 +668,149 @@ class ToggleButtons extends StatelessWidget {
         direction: direction,
         verticalDirection: verticalDirection,
-        child: children[index],
+        child: ClipRRect(
+          borderRadius: clipBorderRadius,
+          child: TextButton(
+            focusNode: focusNodes != null ? focusNodes![index] : null,
+            style: ButtonStyle(
+              backgroundColor: MaterialStateProperty.all<Color?>(effectiveFillColor),
+              foregroundColor: MaterialStateProperty.all<Color?>(currentColor),
+              overlayColor: _ToggleButtonDefaultOverlay(
+                selected:  onPressed != null && isSelected[index],
+                unselected: onPressed != null && !isSelected[index],
+                colorScheme: theme.colorScheme,
+                disabledColor: disabledColor ?? toggleButtonsTheme.disabledColor,
+                focusColor: focusColor ?? toggleButtonsTheme.focusColor,
+                highlightColor: highlightColor ?? toggleButtonsTheme.highlightColor,
+                hoverColor: hoverColor ?? toggleButtonsTheme.hoverColor,
+                splashColor: splashColor ?? toggleButtonsTheme.splashColor,
+              ),
+              elevation: MaterialStateProperty.all<double>(0),
+              textStyle: MaterialStateProperty.all<TextStyle?>(currentTextStyle.copyWith(
+                color: currentColor,
+              )),
+              padding: MaterialStateProperty.all<EdgeInsetsGeometry>(EdgeInsets.zero),
+              minimumSize: MaterialStateProperty.all<Size?>(minimumSize),
+              maximumSize: MaterialStateProperty.all<Size?>(maximumSize),
+              shape: MaterialStateProperty.all<OutlinedBorder>(const RoundedRectangleBorder()),
+              mouseCursor: MaterialStateProperty.all<MouseCursor?>(mouseCursor),
+              visualDensity: VisualDensity.standard,
+              tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+              animationDuration: kThemeChangeDuration,
+              enableFeedback: true,
+              alignment: Alignment.center,
+              splashFactory: InkRipple.splashFactory,
+            ),
+            onPressed: onPressed != null
+              ? () {onPressed!(index);}
+              : null,
+            child: children[index],
+          ),
+        ),
       );
-    });
 
-    final Widget result = direction == Axis.horizontal
-      ? IntrinsicHeight(
-        child: Row(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: buttons,
+      if (currentConstraints != null) {
+        button = Center(child: button);
+      }
+
+      return MergeSemantics(
+        child: Semantics(
+          container: true,
+          button: true,
+          enabled: onPressed != null,
+          child: _InputPadding(
+            minSize: minPaddingSize,
+            direction: direction,
+            child: Semantics(
+              excludeSemantics: true,
+              child: button,
```

Comment:
@TahaTesser this is still off, this additional Semantics widget should not be here. Excluding the semantics of the child leaves it blank without any context.

Reply:
See comment here: https://github.com/flutter/flutter/pull/101760#discussion_r855654325